docs/dev/openapi.yml

```diff
@@ -0,0 +1,481 @@
+      description: The list is returned as a JSON object with an array containing for each entry it's id, title, tags, last visit time and pinned status.
+      responses:
+        200:
+          description: The list of recently viewed notes and pinned notes
+          content:
+            'text/json':
+              schema:
+                type: object
+                properties:
+                  history:
+                    type: array
+                    description: The array that contains history objects
+                    items:
+                      type: object
+                      properties:
+                        id:
+                          type: string
+                          description: The id or alias of the note
+                        text:
+                          type: string
+                          description: The title of the note
+                        time:
+                          type: integer
+                          description: The UNIX-timestamp when the note was last accessed by the user
+                        tags:
+                          type: array
+                          description: The tags that were added by the user to the note
+                          items:
+                            type: string
+                        pinned:
+                          type: boolean
+                          description: Whether the user has pinned this note
+    post:
+      tags:
+        - user
+      summary: Replace user's history with a new one.
+      description: The body must be form-encoded and contain a field `history` with a JSON-encoded array like its returned from the server when exporting the history.
+      requestBody:
+        required: true
+        content:
+          'application/x-www-form-urlencoded':
+            example: 'history=[{"id":"example","text":"Untitled","time":1556275442010,"tags":[],"pinned":false]'
+      responses:
+        200:
+          description: History replaced
+    delete:
+      tags:
+        - user
+      summary: Deletes the user's history.
+      responses:
+        200:
+          description: User's history deleted
+
+  /history/{note}:
+    post:
+      tags:
+        - user
+      summary: Toggles the pinned status in the history for a note.
+      description: The body must be form-encoded and contain a field `pinned` that is either `true` or `false`.
+      requestBody:
+        required: true
+        content:
+          'application/x-www-form-urlencoded':
+            example: 'pinned=false'
+      responses:
+        200:
+          description: Pinned state toggled
+      parameters:
+        - name: note
+          in: path
+          required: true
+          description: The note for which the pinned state should be toggled
+          content:
+            'text/plain':
+              example: my-note
+    delete:
+      tags:
+        - user
+      summary: Deletes a note from the user's history.
```

Comment:
Either this or the rest of this section is wrong, since this talks about note deletion, the rest about pinned notes.

Reply:
I just tested again the requests made by the frontend. A POST request to the endpoint needs the pinned=true/false body, a DELETE request removes the note from the history. Omitting the pinned=true/false body while doing a POST request to the endpoint does not create the history entry but returns a HTTP 500. I'll investigate the code behind this.

Reply:
The methods being called by the history-router are `historyDelete` and `historyPost` in the lib/history.js.

The [`historyPost`](https://github.com/codimd/server/blob/ce99a79352ea145cfbfc579592b96e53ba843b66/lib/history.js#L154-L170) method differentiates between the endpoints with note id and without. In the part for the endpoint with the note id only the toggle for the pinned state will be executed.

The [`historyDelete`](https://github.com/codimd/server/blob/master/lib/history.js#L184-L194) method differentiates similar and in the relevant part for this endpoint it just deletes the note from the user's history.

Reply:
We should really work on a proper `/api/` endpoint collection :see_no_evil: 

Thanks a lot for your investigative work here!

Reply:
Yes we should. But documenting the current status is at least something.